daffodil-runtime1/src/main/scala/org/apache/daffodil/dsom/CompiledExpression1.scala

```diff
@@ -72,6 +75,8 @@ abstract class CompiledExpression[+T <: AnyRef](
   valueForDebugPrinting: AnyRef)
   extends ContentValueReferencedElementInfoMixin with Serializable {
 
+  DataValue.assertValueIsNotDataValue(valueForDebugPrinting)
```

Comment:
This was very useful for doing the refactor. Now that it is done, I am not sure how much this assertion is still needed.